### PR TITLE
test: add lr scheduler tests [DET-5223]

### PR DESCRIPTION
**Pull request by hamidzr**
## Description

- fix a typo in an adapter error message.
- add tests for checking lr goes down monotonically
- add test for lr frequency param in lightning.
- remove explicit num_workers from gan lightning example

## Checklist

- [x] User-facing API changes need the "User-facing API Change" label.
- [x] Release notes should be added as a separate file under `docs/release-notes/`.
See [Release Note](../docs/release-notes/README.md) for details.
- [x] Licenses should be included for new code which was copied and/or modified from any external code.

[DET-1234]: https://determinedai.atlassian.net/browse/DET-1234

### CODE REVIEW

File: harness/determined/pytorch/lightning/_adapter.py [hunk 1]
```diff
@@ -178,9 +180,10 @@ def __init__(self, context: PyTorchTrialContext) -> None:
 
         context.wrap_model(lightning_module)
 
-        pls = _LightningAdapterState(context, lightning_module, [])
+        pls = _LightningAdapterState(context, lightning_module, [], [])
         self._pls = pls
-        optimizers, _ = self.setup_optimizers_schedulers()
+        optimizers, lr_schedulers = self.setup_optimizers_schedulers()
```
Comment (shiyuann) [inline harness/determined/pytorch/lightning/_adapter.py:185]: : D you can do something like this
```suggestion
        pls.optimizers, pls.lr_schedulers = self.setup_optimizers_schedulers()
```

File: harness/tests/experiment/fixtures/lightning_adapter_onevar_model.py [hunk 2]
```diff
@@ -17,23 +18,31 @@ def __getitem__(self, index: int) -> Tuple:
...
 
 class OneVarLM(pl.LightningModule):
-    def __init__(self, *args, **kwargs):
-        super().__init__(*args, **kwargs)
+    def __init__(self, lr_frequency=1, *args, **kwargs):
+        super().__init__()
 
         model = torch.nn.Linear(1, 1, False)
         self.model = model
+        self.save_hyperparameters()
```
Comment (shiyuann) [inline harness/tests/experiment/fixtures/lightning_adapter_onevar_model.py:26]: Why do you need this line?

File: harness/tests/experiment/utils.py [hunk 3]
```diff
@@ -371,6 +371,28 @@ def make_workloads(tag: str) -> workload.Stream:
 ]
 
 
+def train_and_validate(
+    make_trial_controller_fn: Callable[[workload.Stream], det.TrialController],
+    steps: int = 2,
+) -> Tuple[Sequence[Dict[str, Any]], Sequence[Dict[str, Any]]]:
+    metrics: Dict[str, Any] = {"t": [], "v": []}
```
Comment (shiyuann) [inline harness/tests/experiment/utils.py:378]: please make this more readable
```suggestion
    metrics: Dict[str, Any] = {"training": [], "validation": []}
```

File: harness/tests/experiment/pytorch/test_pytorch_lightning.py [hunk 4]
```diff
@@ -87,3 +84,48 @@ def make_trial_controller_fn(
 
         with pytest.raises(AssertionError):
             utils.checkpointing_and_restoring_test(make_trial_controller_fn, tmp_path)
+
+    def test_lr_scheduler(self, tmp_path: pathlib.Path) -> None:
+        class OneVarLAFreq1(la_model.OneVarTrialLRScheduler):
+            def check_lr_value(self, batch_idx: int):
```
Comment (shiyuann) [inline harness/tests/experiment/pytorch/test_pytorch_lightning.py:90]: q: when is this function called?

File: harness/determined/pytorch/lightning/_adapter.py (hunk 1 above)
Comment (hamidzr) [reply harness/determined/pytorch/lightning/_adapter.py:185]: aah I didn't know, that's nicer.

File: harness/tests/experiment/fixtures/lightning_adapter_onevar_model.py (hunk 2 above)
Comment (hamidzr) [reply harness/tests/experiment/fixtures/lightning_adapter_onevar_model.py:26]: to allow this line `                "frequency": self.hparams["lr_frequency"],` 
we could probably do it some other way too but this seems like the standard way of doing it in lightning adatper

File: harness/tests/experiment/pytorch/test_pytorch_lightning.py (hunk 4 above)
Comment (hamidzr) [reply harness/tests/experiment/pytorch/test_pytorch_lightning.py:90]: the parent calls this as part of train_batch
https://github.com/determined-ai/determined/pull/2184/files#diff-1e0d3fc8c227931ea1220947977eb3378f3f12e3916faa15fa74b703d32c7f56R136

File: harness/tests/experiment/fixtures/lightning_adapter_onevar_model.py (hunk 2 above)
Comment (shiyuann) [reply harness/tests/experiment/fixtures/lightning_adapter_onevar_model.py:26]: I see. `save_hyperparameters` always confuses me. But since it's part of the API, it's fine.

**Review by shiyuann — APPROVED** [review]
